driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -635,14 +639,14 @@ public void shutdown() {
             // we're not on an I/O thread or anything, so just call await.
             allChannels.close().awaitUninterruptibly();
 
-            // This will call shutdownNow on the boss and worker executor. Since this is called
-            // only once all connection have been individually closed, it's fine.
-            channelFactory.releaseExternalResources();
+            // This will shut down I/O threads. Since this is called only once all connection have
+            // been individually closed, it's fine.
+            eventLoopGroup.shutdownGracefully().syncUninterruptibly();
```

Comment:
For [JAVA-676](https://datastax-oss.atlassian.net/browse/JAVA-676) won't we not want to do this if the eventLoopGroup was passed in from the outside?  Maybe the EventLoopGroup should be defined outside of the factory and passed in the constructor and shutdown in Cluster.close() if the eventLoopGroup was not provided.  We could also consider sharing EventLoopGroup between all Cluster instances (though I think based on other decisions we've made with executors maybe that isn't a good idea).